ibm_mq/datadog_checks/ibm_mq/ibm_mq.py

```diff
@@ -79,29 +79,36 @@ def check(self, instance):
 
     def discover_queues(self, queue_manager, config):
         queues = []
-        if config.auto_discover_queues:
+        has_patterns = len(config.queue_patterns) > 0
+        if config.auto_discover_queues or has_patterns:
             queues.extend(self._discover_queues(queue_manager, '*'))
-
-        if len(config.queue_patterns) > 0:
-            for regex in config.queue_patterns:
-                queues.extend(self._discover_queues(queue_manager, regex))
```

Comment:
The original implementation did not support regex matching, only MQ wildcard pattern matching.  So this change here really gets at whether `config.queue_patterns` is intended to supply patterns to be treated for regex matching or MQ wildcard pattern matching.  

DataDog folks can decide what the support intent was here.  If the intent was to support MQ wildcard pattern matching only, then this can be adjusted.

That said, having true regex matching on queue names is a `+1` over the less-capable pattern matching that MQ understands natively.

Reply:
Ok, thanks for the explanation